ospf6d/ospf6_intra.c

```diff
@@ -897,83 +951,43 @@ static char *ospf6_intra_prefix_lsa_get_prefix_str(struct ospf6_lsa *lsa,
 static int ospf6_intra_prefix_lsa_show(struct vty *vty, struct ospf6_lsa *lsa,
 				       json_object *json_obj, bool use_json)
 {
-	char *start, *end, *current;
 	struct ospf6_intra_prefix_lsa *intra_prefix_lsa;
 	int prefixnum;
-	char buf[128];
-	struct ospf6_prefix *prefix;
 	char id[16], adv_router[16];
-	struct in6_addr in6;
-	json_object *json_loop;
-	json_object *json_arr = NULL;
-	char prefix_string[133];
+	struct print_prefix_cb_data cbd = { .vty = vty,
+					    .use_json = use_json,
+					    .print_metric = true };
+	struct tlv_handler handler = { .callback = cb_print_prefix,
+				       .callback_data = &cbd };
```

Comment:
additionally will require
```c
.tlv_type = OSPF6_TLV_INTRA_AREA_PREFIX
``` 
for the assert at https://github.com/FRRouting/frr/pull/16199/files#diff-1fba7e8f7310c19c1f1d9996ab826b87c6ea3329ff3383d02e531379b09ebdaeR144
to pass.